### PR TITLE
Adding local-config state

### PR DESCRIPTION
**Pull request by MayuBen1027**
Adding the local-config state based on the inforation found [here](https://github.com/inkdropapp/docs/blob/master/src/pages/reference/state-local-config.md).

### CODE REVIEW

**Comment by Keisir** [comment]
Thanks for the effort.

In its current form, I don't think this page adds any value. We need to check if there is any information we want to display on this page. Otherwise, in my opinion, it makes no difference in having this page.

Maybe Takuya has a hint what could be added here. I couldn't find anything in the type definitions.
https://github.com/inkdropapp/types/blob/main/packages/%40inkdropapp/redux-store/reducers/config.d.ts

**Comment by craftzdog** [comment]
This state allows you to use config values of `config.json` in your react components.
It immediately reflects to the state whenever the config file is changed.
The data structure is same as the file content.
The type def currently looks like so:

```ts
import type { ValidateFunction } from 'ajv'
import ConfigSchema from '../json-schema/config.json'
import validator from '../validators/config'
export type NoteListSortBy = 'title' | 'updatedAt' | 'createdAt' | 'rank'
export type NoteListOrder = 'asc' | 'desc'
export type TagListOrder = 'name' | 'count'
export type Config = {
  _id: 'config'
  _rev?: string
  core: {
    defaultBook?: string
    noteListOrder?: NoteListOrder
    noteListSortBy?: NoteListSortBy
    tagListOrder?: TagListOrder
    queryOptions?: Record<
      string,
      {
        noteListOrder?: NoteListOrder
        noteListSortBy?: NoteListSortBy
      }
    >
  }
  updatedAt: number
}
const validateConfig: ValidateFunction<Config> = validator
export { ConfigSchema, validateConfig }
```

But I think we should cover the config module first: https://docs.inkdrop.app/reference/config

File: src/app/states/local-config/page.mdx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+export const metadata = {
+  title: 'local-config',
+  description: 'An app state of the local config. This section explains how to fetch the `local config` state.',
```
Comment (craftzdog) [inline src/app/states/local-config/page.mdx:3]: ❌ `local config`
⭕️ `localConfig`

File: src/app/states/local-config/page.mdx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+You can obtain the `local config` state and integrate it with React components.{{ className: 'lead' }}
+
+
+---
+
+## Data Structure
+
+<Row>
+  <Col sticky>
+    Get the `local config` state manually:
```
Comment (craftzdog) [inline src/app/states/local-config/page.mdx:18]: ❌ local config
⭕️ localConfig

File: src/app/states/local-config/page.mdx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    ```js
+    const { config } = inkdrop.store.getState()
```
Comment (craftzdog) [inline src/app/states/local-config/page.mdx:21]: The doc was incorrect. It should be

```
const { localConfig } = inkdrop.store.getState()
```

File: src/app/states/local-config/page.mdx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    ```
+
+    Connect the state with your React component:
+
+    ```js
+    import { useSelector } from 'react-redux'
+    const selector = ({ config }) => config
```
Comment (craftzdog) [inline src/app/states/local-config/page.mdx:28]:
```
const selector = ({ localConfig }) => localConfig
```

**Review by craftzdog — CHANGES_REQUESTED** [review]

**Comment by craftzdog** [comment]
I guess the description of the data structure of the local config will not be so useful for developers because they are basically for the internal use.
It would be useful for plugins that have their own config values like:

```js
inkdrop.config.set('my-plugin.anInt', '123')
```

In this case, you can use `localConfig` like so:

```js
import { useSelector } from 'react-redux'
const selector = ({ localConfig }) => localConfig['my-plugin']
```

**Pull request closed without merging**